packages/common/src/providers/httpProvider.ts

```diff
@@ -32,7 +32,7 @@ export class HttpProvider implements Provider {
                 body: JSONBig.stringify(request),
                 headers: {
                     'Content-Type': 'application/json',
-                    cookie: this.cookie,
+                    ...(this.cookie && { cookie: this.cookie }),
```

Comment:
This seems weird for me, but it seems to be a legitimate conditional concat 🤷

Would be easier to read like this though IMO:

```ts
const headers = {...};

if (this.cookie !== undefined) {
    headers.cookie = cookie;
}

...
```

If you prefer the current form just ignore this 😉